bimmer_connected/api/authentication.py

```diff
@@ -270,14 +274,33 @@ async def _login_china(self):
             encrypted = cipher_rsa.encrypt(self.password.encode())
             pw_encrypted = base64.b64encode(encrypted).decode("UTF-8")
 
-            cipher_aes = AES.new(**get_aes_keys(self.region), mode=AES.MODE_CBC)
-            nonce = f"{self.username}|{datetime.datetime.utcnow().strftime('%Y-%m-%d %H:%M:%S.%fZ')}".encode()
+            captcha_res = await client.post(
+                AUTH_CHINA_CAPTCHA_URL,
+                json={"mobile": self.username},
+            )
+            verify_id = captcha_res.json()["data"]["verifyId"]
+
+            for i in range(1, 13):
+                try:
+                    captcha_check_res = await client.post(
+                        AUTH_CHINA_CAPTCHA_CHECK_URL,
+                        json={"position": 0.74 + i / 100, "verifyId": verify_id},
+                    )
+                    if captcha_check_res.json()["code"] == 200:
+                        break
+                except Exception:
```

Comment:
Assuming the exception is raised directly from the client because of an HTTP 400 status code, we should not use bare `except Exception` but rather `except httpx.HTTPStatusError`.

Reply:
I have already tried, but I cannot obtain `httpx.HTTPStatusError` here because the error is caught and handled by the client's `raise_for_status_event_handler`.

Reply:
Ah, my bad. You should be able to catch `MyBMWAPIError` here. Maybe combine with request status like this:
```python
except MyBMWAPIError as ex:
    if captcha_check_request.status_code == 429:
       continue
    else:
        raise ex
```

Reply:
Now I'm catching `MyBMWAPIError`, but I can't access the current `captcha_check_response` within the `except` block, and the `ex` only raises `error_message` with `_ex_to_raise(_err_message)`. Since the scope of the try block is only for attempting the verification request, any errors caught can be silently skipped.